src/models/json-api-query-data.ts

```diff
@@ -0,0 +1,14 @@
+import {JsonApiModel} from './json-api.model';
+
+export class JsonApiQueryData {
```

Comment:
can you make this class generic too? thus the need for a typecast when using getModels() would be obsolete.
Besides of this little change the PR LGTM ;)